File: ObjectAvoidance/java/src/main/java/elite/math/trig/Tangent.java
package elite.math.trig;

import elite.math.trig.triangle.Adjacent;
import elite.math.trig.triangle.Opposite;

/**
 *
 * @author CFerg (Elite)
 */
public class Tangent {

    /**
     *
     */
    public double tangent;
    
    /**
     *
     * @param o
     * @param a
     */
    public Tangent(Opposite o, Adjacent a){
        this.tangent = (o.opposite / a.adjacent);
    }
    
    /**
     *
     * @param arctan
     */
    public Tangent(Arctangent arctan){
        this.tangent = (1 / arctan.arctangent);
    }
    
    /**
     *
     * @param cot
     */
    public Tangent(Cotangent cot){
        this.tangent = (1 / cot.cotangent);
    }
}

File: ObjectAvoidance/src/elite/math/trig/Tangent.kt
package elite.math.trig

import elite.math.trig.triangle.Adjacent
import elite.math.trig.triangle.Opposite

/**
 *
 * @author CFerg (Elite)
 */
class Tangent {

    /**
     *
     */
    var tangent: Double = 0.toDouble()

    /**
     *
     * @param o
     * @param a
     */
    constructor(o: Opposite, a: Adjacent) {
        this.tangent = o.opposite / a.adjacent
    }

    /**
     *
     * @param arctan
     */
    constructor(arctan: Arctangent) {
        this.tangent = 1 / arctan.arctangent
    }

    /**
     *
     * @param cot
     */
    constructor(cot: Cotangent) {
        this.tangent = 1 / cot.cotangent
    }
}